holoviews/plotting/mpl/chart.py

```diff
@@ -1065,18 +954,20 @@ def get_data(self, element, ranges):
         if cdim:
             array = element.dimension_values(cdim)
             clim = ranges[cdim.name]
-        return data, array, clim
+        style['array'] = array
+        style['clim'] = clim
+        return [np.array(data)], style, {}
 
 
-    def update_handles(self, axis, element, key, ranges=None):
+    def update_handles(self, key, axis, element, ranges, style):
         artist = self.handles['artist']
-        data, array, clim = self.get_data(element, ranges)
+        (data,), kwargs, axis_kwargs = self.get_data(element, ranges, style)
```

Comment:
It's the arguments to the plotting call and you're right they should be consistent.